ts/util/getInitials.ts

```diff
@@ -1,24 +1,38 @@
 // Copyright 2018-2020 Signal Messenger, LLC
 // SPDX-License-Identifier: AGPL-3.0-only
 
-const BAD_CHARACTERS = /[^A-Za-z\s]+/g;
-const WHITESPACE = /\s+/g;
-
-function removeNonInitials(name: string) {
-  return name.replace(BAD_CHARACTERS, '').replace(WHITESPACE, ' ');
-}
-
 export function getInitials(name?: string): string | undefined {
   if (!name) {
     return undefined;
   }
 
-  const cleaned = removeNonInitials(name);
-  const parts = cleaned.split(' ');
-  const initials = parts.map(part => part.trim()[0]);
-  if (!initials.length) {
+  const parsedName = name
+    // remove all chars that are not letters or separators
+    .replace(/[^\p{L}\p{Z}]+/gu, '')
+    // replace all chars that are separators with a single ASCII space
+    .replace(/\p{Z}+/gu, ' ')
+    // remove "Jr", "Sr" at the end of the string
+    .replace(/\s+(jr|sr)$/gi, '')
+    // remove Roman numerals at the end of the string
+    .replace(
+      /\s(?=[MDCLXVI])M*(C[MD]|D?C{0,3})(X[CL]|L?X{0,3})(I[XV]|V?I{0,3})$/gi,
+      ''
+    )
```

Comment:
It probably makes sense for a lot of English names, but I could imagine some false positives in other languages. Could we remove these?

Reply:
Good point! Removed.